src/marqo/tensor_search/add_docs.py

```diff
@@ -0,0 +1,48 @@
+"""Functions used to fulfill the add_documents endpoint"""
+import copy
+import functools
+import math
+import threading
+import warnings
+from typing import List
+import PIL
+from marqo.s2_inference.clip_utils import _is_image, load_image_from_path
+
+
+def threaded_download_images(allocated_docs: List[dict], image_repo: dict) -> None:
+    """A thread calls this function to download images for its allocated documents
+
+    This should be called only if treat URLs as images is True
+    """
+    for doc in allocated_docs:
+        for field in list(doc):
+            if isinstance(doc[field], str) and _is_image(doc[field]):
+                try:
+                    image_repo[doc[field]] = load_image_from_path(doc[field])
```

Comment:
would this not cause overwrites? for example if Doc 1 and Doc 2 both have "Field A", they could both save their images under `image_repo["Field A"]`

Reply:
`image_repo[doc[field]]` - the key of image_repo is the value associated with the `field`  in the doc dict. This would be a URL. So the image_repo would have this structure: 
```
{
    "https://s3.aws.com/my_image1.png": <PIL_image>, 
    "https://s3.aws.com/my_image2.png": <PIL_image> 
}
```

Reply:
ah okay, got it.